modules/tinymce/src/core/main/ts/fmt/MergeFormats.ts

```diff
@@ -12,7 +12,7 @@ const each = Tools.each;
 
 const mergeTextDecorationsAndColor = (dom: DOMUtils, format: ApplyFormat, vars: FormatVars | undefined, node: Node): void => {
   const processTextDecorationsAndColor = (n: Node) => {
-    if (NodeType.isElement(n) && NodeType.isElement(n.parentNode)) {
+    if (NodeType.isElement(n) && NodeType.isElement(n.parentNode) && dom.getContentEditable(n.parentNode) !== 'false' && dom.getContentEditable(n) !== 'false') {
```

Comment:
Hmm... this logic doesn't look right. Wouldn't this prevent merging formats if you were only applying it inside a `contenteditable="true"` container (see below) either that or wouldn't it fail to walk up if nested a few levels deep? I think instead we need to check they are within the same contenteditable block.

```html
<p contentediable="false">noneditable <span contenteditable="true"><span style="color: red">editable</span> text</span></p>
```

Reply:
I have simplified the noneditable check a bit for this function so I think it is all good now